optimizer_wrapper.rb

```diff
@@ -585,31 +622,36 @@ def self.compute_result_total_dimensions_and_round_route_stats(result)
   end
 
   def self.compute_route_waiting_times(route)
-    seen = 1
-    previous_end =
-      if route[:activities].first[:type] == 'depot'
-        route[:activities].first[:begin_time]
+    previous_end = route[:activities].first[:begin_time]
```

Comment:
Why is the check of first_route being a depot not needed anymore ? I feel route[:activities].first[:begin_time] is only valid if first activity is a depot. If it is more homogeneous now, shouldn't we now use route[:activities].first[:end_time] ?

Reply:
A route always have a start_time if we have the time matrix.
Regardless, the type of the first activity, this first computation represent this start_time.

Here, it cannot be end_time to ensure the correct computation of the consumed_setup_time variable for the first activity, especially if the first activity is a rest and the next service has a setup_duration.

Reply:
Maybe there is something I do not understand. It seems to me that in the case first route activity is not a depot then we do not consider this first activity duration in l.651 (arrival_time computation).

Reply:
If we have as first activity a rest and then a service with a setup_duration.

In some situation setup_duration may be applied before the rest. Then if i do not compute correctly this value from the beginning, we may have a shift in the time computed. Or we will have a wrongly computed waiting_time or we will consider twice the setup_duration

Reply:
I agree with the fact that you recompute the value. I just do not see the case where first activity is a visit. (And I am wondering if part of your previous answer did not disappear ?)